### ISSUE TITLE
[bug]: 

### ISSUE THREAD

**Issue by KalervoHyyppa** [pmndrs/react-spring#2004]
### Which react-spring target are you using?

- [ ] `@react-spring/web`
- [ ] `@react-spring/three`
- [ ] `@react-spring/native`
- [ ] `@react-spring/konva`
- [ ] `@react-spring/zdog`

### What version of react-spring are you using?

"@react-spring/parallax": "^9.5.5"

### What's Wrong?

Running `gatsby build` causes error:

```
  76 |       translate = start * parent.space;
  77 |     } else {
> 78 |       const targetScroll = Math.floor(offset) * parent.space;
     | ^
  79 |       const distance = parent.space * offset + targetScroll * speed;
  80 |       translate = -(parent.current * speed) + distance;
  81 |     }


  WebpackError: TypeError: Cannot read property 'space' of null
```

Seems to only occur when using `<Parallax/>` and `<ParallaxLayer/>` in separate files.

I've tried passing in a ref into the child component using `<ParallaxLayer/>`, but the error still occurs

Runs fine with `gatsby develop`

## Environment

- `react-spring/parallax` v9.5.5 
- `react` v18.1.0
- `gatsby` v4.24.4

### To Reproduce

Run`gatsby build`

### Expected Behaviour

Build without errors.

### Link to repo

https://github.com/KalervoHyyppa/gatsby_example_parallax

**Comment by joshuaellis** [comment]
This is down to how `gatsby` builds it's pages, it seems to do reverse order(?) so the Context is not initiated before the child is:

<img width="179" alt="CleanShot 2022-10-25 at 19 23 36@2x" src="https://user-images.githubusercontent.com/37798644/197852042-22ee7a15-b994-4219-b538-3b09ec275c16.png">

So i'm not sure even if you could get it to build that it would work as intended. You could look at fixing this in a local version of the repo (you would have to add checks to see if `parent` is defined but I'm still not sure it would work.

**Comment by KalervoHyyppa** [comment]
Good catch. I just moved `Page 1` and `Page 2` into a different folder (components in this case) and it worked. Updated the repo as well for context. Will close this issue now.
